modules/core/shared/src/main/scala/lucuma/core/model/sequence/Step.scala

```diff
@@ -86,17 +95,117 @@ object Step extends WithUid('s') {
       Focus[GmosSouth](_.breakpoint)
   }
 
-  implicit val eqStep: Eq[Step] = Eq.instance {
+  implicit val eqFutureStep: Eq[FutureStep] = Eq.instance {
     case (a @ GmosNorth(_, _, _, _, _), b @ GmosNorth(_, _, _, _, _)) => a === b
     case (a @ GmosSouth(_, _, _, _, _), b @ GmosSouth(_, _, _, _, _)) => a === b
     case _                                                            => false
   }
 
   /** @group Optics */
-  val gmosNorth: Prism[Step, GmosNorth] =
-    GenPrism[Step, GmosNorth]
+  val gmosNorth: Prism[FutureStep, GmosNorth] =
+    GenPrism[FutureStep, GmosNorth]
+
+  /** @group Optics */
+  val gmosSouth: Prism[FutureStep, GmosSouth] =
+    GenPrism[FutureStep, GmosSouth]
+}
+
+sealed trait StepRecord extends Step {
+  def created: Instant
+  def startTime: Option[Instant]
+  def endTime: Option[Instant]
+
+  def duration: Option[Duration] =
+    (startTime, endTime).mapN(Duration.between)
+}
+
+object StepRecord {
+  final case class GmosNorth(
+    id:               Step.Id,
+    created:          Instant,
+    startTime:        Option[Instant],
+    endTime:          Option[Instant],
+    instrumentConfig: DynamicConfig.GmosNorth,
+    stepConfig:       StepConfig
+  ) extends StepRecord
```

Comment:
I'm not sure it matters for the purpose of modeling this, but as I mentioned in a separate email the times come from (and must be consistent with) step events from Observe.

Reply:
I see. I'll add the step events then and remove the start and end times.